src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -439,6 +480,34 @@ private static object CreateInstance([DynamicallyAccessedMembers(DynamicallyAcce
             return instance ?? throw new InvalidOperationException(SR.Format(SR.Error_FailedToActivate, type));
         }
 
+        private static bool DoAllParametersHaveEquivalentProperties(ParameterInfo[] parameters,
+            List<PropertyInfo> properties, out string s)
+        {
+            var parameterNames = parameters.Select(param => param.Name).ToList();
```

Comment:
Can we get rid of this Linq usage? We try not using it as much as possible in dotnet/runtime. (Although ConfigurationBinder code is already filled with inefficiencies and allocations.)